lib/portlayer/exec/exec.go

```diff
@@ -161,69 +155,20 @@ func eventCallback(ie events.Event) {
 				}()
 			case StateRemoved:
 				log.Debugf("Container(%s) %s via event activity", container.ExecConfig.ID, newState.String())
+				if container.vm != nil && container.vm.Fixing {
+					// is fixing vm, do not remove from containers
+					log.Debugf("Container(%s) %s is being fixed", container.ExecConfig.ID)
+					container.vm.Fixing = false
```

Comment:
You should use the golang [atomic](https://golang.org/pkg/sync/atomic/) package here instead of setting this field directly.